### PR TITLE
Extract into ApkInvoker and split into data classes.

### PR DESCRIPTION
**Pull request by sv99**
Sorry for massive changes!

1. extract new class AaptInvoker
2. rename MetaFile to ApkInfo and rename meta namespace to apk
3. rename AndrolibResources to the ResourcesDecoder - name unification
4. rename private config to mConfig
5. extract internal classes ARSCData and FlagsOffset from the ARSCDecoder
6. move decodeManifest and decodeResources to the ResourcesDecoder
7. minimize public API in all decoder classes
8. move you data classes to data namespace

rebased

### CODE REVIEW

**Comment by iBotPeaches** [comment]
okay give me a few days on this one :)

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/ApkDecoder.java [hunk 1]
```diff
@@ -68,8 +61,8 @@ public ApkDecoder(ExtFile apkFile) {
     }
 
     public ApkDecoder(Config config, ExtFile apkFile) {
-        this.config = config;
-        mAndRes = new AndrolibResources(config);
+        mConfig = config;
+        //mAndRes = new AndrolibResources(config);
```
Comment (iBotPeaches) [inline brut.apktool/apktool-lib/src/main/java/brut/androlib/ApkDecoder.java:65]: Unneeded now?

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/apk/ApkInfo.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class ApkInfo {
+    public String version;
+
+    private String mApkFileName;
+    public boolean isFrameworkApk;
+    public UsesFramework usesFramework;
+    private  Map<String, String> mSdkInfo = new LinkedHashMap<>();
```
Comment (iBotPeaches) [inline brut.apktool/apktool-lib/src/main/java/brut/androlib/apk/ApkInfo.java:41]: very minor - 2 spaces

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/res/ResourcesDecoder.java [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+
+final public class ResourcesDecoder {
```
Comment (iBotPeaches) [inline brut.apktool/apktool-lib/src/main/java/brut/androlib/res/ResourcesDecoder.java:43]: I'm debating in my head if we should call this `ResourceDecoder` (singular)

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/res/ResourcesDecoder.java [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+
+    private final static String[] APK_RESOURCES_FILENAMES = new String[] {
+        "resources.arsc", "res", "r", "R" };
+    private final static String[] APK_MANIFEST_FILENAMES = new String[] {
+        "AndroidManifest.xml" };
+    private final static String[] IGNORED_PACKAGES = new String[] {
+        "android", "com.htc", "com.lge", "com.lge.internal", "yi", "flyme", "air.com.adobe.appentry",
+        "FFFFFFFFFFFFFFFFFFFFFF" };
+
+
```
Comment (iBotPeaches) [inline brut.apktool/apktool-lib/src/main/java/brut/androlib/res/ResourcesDecoder.java:61]: Can we only have 1 newline here?

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/apk/ApkInfo.java (hunk 2 above)
Comment (sv99) [reply brut.apktool/apktool-lib/src/main/java/brut/androlib/apk/ApkInfo.java:41]: In which place?

Comment (iBotPeaches) [reply brut.apktool/apktool-lib/src/main/java/brut/androlib/apk/ApkInfo.java:41]: `private  Map`
vs
`private Map`

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/Config.java [hunk 5]
```diff
@@ -48,12 +47,9 @@ public class Config {
     public boolean verbose = false;
     public boolean copyOriginalFiles = false;
     public boolean updateFiles = false;
-    public boolean isFramework = false;
-    public boolean resourcesAreCompressed = false;
     public boolean useAapt2 = false;
     public boolean noCrunch = false;
     public int forceApi = 0;
-    public Collection<String> doNotCompress;
```
[the remark this answers is not in the document]
Comment (sv99) [reply brut.apktool/apktool-lib/src/main/java/brut/androlib/Config.java:53]: This is not configuration parameters.
This is apk properties.

Config store params from command line.
ApkInfo store apk properties.

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/apk/ApkInfo.java (hunk 2 above)
Comment (sv99) [reply brut.apktool/apktool-lib/src/main/java/brut/androlib/apk/ApkInfo.java:41]: Ok

**Review by iBotPeaches — APPROVED** [review]
Found some minor things I'll clean up in a follow up PR. Would rather get this merged and fix a few things then hold up a 40 file PR.

**Pull request merged**